applicatie/templates/macros.html

```diff
@@ -143,89 +143,126 @@ <h4>Uitvoer van {{ resultaat.controle.omschrijving }}</h4>
-{% set x = kolomkeuzes.update({"Rij_code":    ["code_1", "code_2"]}) %}
-{% set x = kolomkeuzes.update({"Kolom_soort": ["categorie", "stand_per"]}) %}
-{% set x = kolomkeuzes.update({"Kolom_code":  ["code_3", "code_4"]}) %}
-{% set x = kolomkeuzes.update({"Bedrag":      []}) %}
-{% set x = kolomkeuzes.update({"Opmerking":   []}) %}
+{% set formulier_kolommen = ["Type", "Rij_soort", "Rij_code", "Kolom_soort", "Kolom_code"] %}
+{% set formulier_kolommen = formulier_kolommen + ["Bedrag", "Opmerking", "Reset", "Check"] %}
+
+{% set type_opties = {"Type": ["lasten", "baten", "balans_lasten", "balans_baten", "balans_standen"]} %}
+
+{% set rij_soort_opties = {} %}
+{% set rij_code_opties = {} %}
+{% set kolom_soort_opties = {} %}
+{% set kolom_code_opties = {} %}
+
+{# haal de opties op voor de form-controls #}
+{# rij_soort, rij_code, kolom_soort, kolom_code #}
+
+{% for item in type_opties["Type"] %}
+    {% set foo = rij_soort_opties.update({item: [draaitabel[item].rij_naam.title().lower()]}) %}
+    {% set bar = rij_code_opties.update({item: draaitabel[item].geef_rij_keuzes()}) %}
+    {% set foo = kolom_soort_opties.update({item: [draaitabel[item].kolom_naam.title().lower()]}) %}
+    {% set bar = kolom_code_opties.update({item: draaitabel[item].geef_kolom_keuzes()}) %}
+{% endfor %}
+
 
 <form class="form-inline text-center muteerknoppen"
+      id="Form_bulkcorrecties"
       role="form"
       method="post"
 >
+
     <div class="form-group" style="margin-bottom: 20px;float: center">
-        <input class="form-control mutatie-invoer" readonly="readonly"
+        <input class="form-control mutatie-invoer"
+               readonly="readonly"
                style="color: blue;width: 600px;margin-right: 200px;text-align: center;"
                value="Voer hieronder de correcties in en klik op de knop 'Correcties verwerken'">
 
-        <button class="btn btn-primary" style="float: right" type="button"
+        <button class="btn btn-primary" style="margin-right: 20px;"
+                type="button"
                 title="Klik om correcties te verwerken"
-                onclick="alert('Deze knop werkt nog niet!');">
+                onclick="bulkcorrecties_verwerken();">
                 Correcties verwerken
         </button>
-    </div>
 
-    {% for n in range(aantal_rijen) %}
+        <button class="btn btn-primary" style="float: right"
+                type="button"
+                title="Klik om formulier leeg te maken"
+                onclick="formulier_bulkcorrecties_leegmaken();">
+                Formulier leegmaken
+        </button>
+    </div>
 
-    <div style="margin-bottom: 4px">
+    {% for rij in range(aantal_rijen) %}
 
-        {% for kol, type in kolommen.items() %}
+    <div style="margin-bottom: 4px" id="{{ 'bulkcorr_' + rij|string }}">
 
-        <div class="form-group" style="padding-right: 10px;" id="{{ 'corr_' + n|string }}">
-            {% if n == 0 %}
+        {% for kol in formulier_kolommen %}
```

Comment:
ik zou kolom gewoon voluit typen ipv kol, dat is duidelijker. Ik denk dat afkortingen voor kleine dingen als: 
`for k in kolommen: 
print(k)`
 wel kan, maar als je k op meerdere plekken gaat gebruiken wordt het duidelijker als k gewoon kolom heet.

Reply:
OK, zal ik aanpassen.